### DISCUSSION TITLE
Should we release 0.69.2?

### DISCUSSION THREAD

**Discussion by fortmarek** [reactwg/react-native-releases#24] [Patches]
# Released.
Please refer to https://github.com/reactwg/react-native-releases/discussions/27 for follow ups

---

## Should we release 0.69.2?

Current Release: [0.69.1](https://github.com/facebook/react-native/releases/tag/v0.69.1)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

#### List of qualified picks

- https://github.com/facebook/react-native/commit/0bd523955385a3b1e622077b7ee4ea0df3c5c158, https://github.com/facebook/react-native/commit/9f112af50abbcd059e3872ae63143c35bb8c0f7f (to be combined to a single PR for 0.69 specifically):
  - https://github.com/facebook/react-native/pull/34098 (hence, already in 0.69 branch)
- https://github.com/facebook/react-native/commit/2fb6a3393d545a93518d1b2906bd9453458660a0
- https://github.com/facebook/react-native/commit/b5ff26b0b97d6cd600bdb9c33af866971ef14f9c
- https://github.com/facebook/react-native/commit/f97c6a5b498eec95e99a02c7842cb2ae160cd6cd
- https://github.com/facebook/react-native/commit/bd12e41188c8d85c0acbd713f10f0bd34ea0edca
- https://github.com/facebook/react-native/commit/4ea38e16bf533955557057656cba5346d2372acd (might be already in 0.69.1?)
- https://github.com/facebook/react-native/pull/34104
- https://github.com/facebook/react-native/pull/34159

#### Local commits to backport to main

1.

**Comment by geraintwhite** [comment]
Can we cherry-pick the fix for jest 28 support? https://github.com/facebook/react-native/commit/b5ff26b0b97d6cd600bdb9c33af866971ef14f9c

**Reply by ginnymin** [reply]
+1 to this, I came here to ask the same!

**Comment by mikehardy** [comment]
We have some trouble brewing with the patch from @kudo to fix the `use_frameworks!` + `hermes_enabled = true` case, reported by @raghavyadavm (thank you!):

https://github.com/facebook/react-native/pull/34030#issuecomment-1170459454

it only shows up when you go turn it on a real device and sadly my qualification runs were done against Simulators :disappointed: 

[Updated: tested other scenarios. **It's not a regression from 0.69.0, it is just a regression of expectations** - we thought it would work - 0.69.0 did not work for use_frameworks at all, and 0.69.1 does not either]

~~This *may* be an issue that causes total failure on iOS builds to real devices on default configurations, so wanted to put a heads up here as a revert + 0.69.2 might be appropriate if someone else confirms the failure. At the very least~~ it appears to cause failures to deploy on real iOS device for any `use_frameworks!` build


~~I'm checking other scenarios and will propose a revert PR for this against main (for cherry-pick here) as soon as I can since~~ my first quick attempts at a repair were ineffective. Someone with more objective-c / cocoapods knowledge may know how to repair this really easily though of course, that would be much better and I'll happily see my revert PR closed

workarounds:

- If you disable bitcode, you can use_frameworks and have hermes either on or off, which is cool ~~but disabling bitcode is a shame.~~ and [disabling bitcode will be the default in Xcode 14](https://stackoverflow.com/questions/72543728/xcode-14-deprecates-bitcode-but-why/72552590) so this is reasonable
- If you do a patch package that reverts https://github.com/facebook/react-native/pull/34030 + remove ios/Pods + pod install then you can do use_frameworks again 0.69.1, but no hermes

**Reply by cortinico** [reply]
Is this also causing issues like this one? https://github.com/facebook/react-native/issues/34102

**Reply by mikehardy** [reply]
@cortinico not sure but I assume you post that here because you have a hunch it does - reading that issue it is the *exact* area being altered by the other related patch - https://github.com/facebook/react-native/pull/34011 - but the "steps to reproduce" being really vague on that one and the multi-splat of screenshots vs text mean I'm not sure exactly. I won't have time to pursue either unfortunately, have to set react-native "integration / release-tester" work aside for a bit as I'm set for a batch of travel

**Comment by brunoducattii** [comment]
Hi. I followed every step to update but until without success:

**What’s Wrong?** 
 App is crashing after upgrade to RN.0.69.1, new arch enable by RCT_NEW_ARCH_ENABLED=1 pod install.
-  Emulating on Xcode = stucked in splashscreen
-  Archive/TestFligh = crash after opening 

**Error**
<img width="986" alt="Screen Shot 2022-07-01 at 09 53 16" src="https://user-images.githubusercontent.com/65225461/176898806-3657de6d-f451-488f-ba5e-4627ba67aebf.png">

**NPX info**
System:
    OS: macOS 12.4
    CPU: (8) arm64 Apple M1
    Memory: 92.42 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
    Yarn: 1.22.19 - ~/Desktop/myapp/app/node_modules/.bin/yarn
    npm: 6.14.17 - ~/.nvm/versions/node/v16.15.1/bin/npm
    Watchman: 2022.03.21.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK:
      Android NDK: 24.0.8215888
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8512546
    Xcode: 13.4/13F17a - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.14 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0
    react-native: 0.69.1 => 0.69.1
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

**Reply by Arkkeeper** [reply]
I guess, your terminal tells something like "new NativeEventEmitter() was called with a non-null argument" at this time, right?

**Comment by kelset** [comment]
important one to not forget: we need a new CLI v8 release to fix the warn issue: https://github.com/reactwg/react-native-releases/discussions/23#discussioncomment-3006097

**Reply by kelset** [reply]
@thymikee any updates on this?

**Reply by thymikee** [reply]
Working on it today, will keep it updated

**Reply by thymikee** [reply]
PR is here: https://github.com/react-native-community/cli/pull/1640

**Reply by thymikee** [reply]
Released v8.0.3. Is anyone able to help me test it? You'll need to [regenerate the lock file entries regarding `@react-native-community/cli` packages](https://github.com/react-native-community/cli#updating-the-cli): cc @AlexanderEggers @swrobel @tgensol.

PR to RN 0.69: https://github.com/facebook/react-native/pull/34159

**Reply by kelset** [reply]
Great stuff @thymikee! Thanks for working on this - sidequestion, is this change also on CLIv9?

**Reply by thymikee** [reply]
@kelset yes, alongside autolinking adjustments from @cortinico. The version is 9.0.0-alpha.3 and is PR'd here https://github.com/facebook/react-native/pull/34160

**Reply by swrobel** [reply]
@thymikee I'm getting the following error now:
```
$ yarn list --pattern @react-native-community/cli
yarn list v1.22.19
├─ @react-native-community/cli-clean@8.0.0
├─ @react-native-community/cli-config@8.0.3
├─ @react-native-community/cli-debugger-ui@8.0.0
├─ @react-native-community/cli-doctor@8.0.3
├─ @react-native-community/cli-hermes@8.0.2
├─ @react-native-community/cli-platform-android@8.0.2
├─ @react-native-community/cli-platform-ios@8.0.2
├─ @react-native-community/cli-plugin-metro@8.0.0
├─ @react-native-community/cli-server-api@8.0.0
├─ @react-native-community/cli-tools@8.0.0
├─ @react-native-community/cli-types@8.0.0
└─ @react-native-community/cli@8.0.3
✨  Done in 0.33s.

$ yarn ios
yarn run v1.22.19
$ react-native run-ios
error Failed to load configuration of your project.
Error: Cannot find module 'node_modules/@fortawesome/free-brands-svg-icons/package.json.js'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:960:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:953:15)
    at resolveExports (node:internal/modules/cjs/loader:482:14)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.resolve (node:internal/modules/cjs/helpers:108:19)
    at resolveNodeModuleDir (node_modules/@react-native-community/cli-tools/build/resolveNodeModuleDir.js:24:42)
    at node_modules/@react-native-community/cli-config/build/loadConfig.js:93:76
    at Array.reduce (<anonymous>)
    at loadConfig (node_modules/@react-native-community/cli-config/build/loadConfig.js:91:134)
info Run CLI with --verbose flag for more details.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Reply by thymikee** [reply]
@swrobel this looks like an issue with node's ESM resolver. Like something is appending a `.js` extension to `require.resolve` (and hence requiring `package.json` results in `package.json.js`). Can you share something more about your setup?

**Reply by swrobel** [reply]
@thymikee not sure what would be useful to share, but with the same app on RN 0.67.4, I don't have any such issue. This only has come up since attempting the upgrade to 0.69.1.

```
$ node -v
v16.16.0
```

**Reply by thymikee** [reply]
I was able to repro that with this specific package, investigating now.

**Reply by thymikee** [reply]
Looks like it's related to this change in fortawesome v6: https://github.com/FortAwesome/Font-Awesome/blame/6.x/js-packages/%40fortawesome/free-brands-svg-icons/package.json#L50. Now that exports are defined explicitly, node can't resolve package.json, so it appends the `.js` as instructed by the ESM `exports` config. Not sure what was the idea behind it and whether this is something becoming a standard in the node.js community or not. This is the first time I encounter this. There are lots of other packages that query for `<package-name>/package.json` because it was always guaranteed to be there, so I assume it's a bug in the configuration. Please contact the library maintainers

**Reply by kelset** [reply]
@thymikee @swrobel looks like they are already aware: https://github.com/FortAwesome/Font-Awesome/issues/18844#issuecomment-1177819547

**Reply by swrobel** [reply]
@thymikee @kelset thank you both for your quick responses, and my apologies for sending you down that rabbit hole - I forgot that I had upgraded my FontAwesome deps along with my RN version, so I wasn't comparing the two accurately. Upon reverting to the same versions I was using with RN 0.67.4, and using the updated RN 0.69.1 + cli 8.0.3 I'm now able to build my app. Thank you @thymikee for the fix!

**Comment by wood1986** [comment]
Hey is it possible to include this tiny bug fix in 0.69.2?

https://github.com/facebook/react-native/pull/34096

**Reply by kelset** [reply]
we can try! link to commit: https://github.com/facebook/react-native/commit/bd12e41188c8d85c0acbd713f10f0bd34ea0edca

**Comment by kelset** [comment]
Another commit: https://github.com/facebook/react-native/commit/4ea38e16bf533955557057656cba5346d2372acd

**Reply by thomazcapra** [reply]
Is the iOS 11 still supported after this update?

![image](https://user-images.githubusercontent.com/13137325/177623958-e7080ad4-d39b-4372-b38e-c1f4c28541bb.png)

**Reply by kelset** [reply]
@thomazcapra no it's not: https://reactnative.dev/blog/2022/06/21/version-069#highlights-of-069

**Comment by AfrazHussain** [comment]
https://github.com/facebook/react-native/issues/34102 is causing quite a lot of issues with existing libraries. There seems to be something wrong with React-bridging paths, and it only happens when you add a few external libraries - the one is the issue is `react-native-vision-camera` and the one that I have tested is `@shopify/react-native-skia`. Steps to reproduce and a few suggestions are mentioned in the issue. 

P.S. I know I'm not cherry picking a PR but I'm just bringing it to everyone's attention for the 0.69.2 release.

**Reply by kelset** [reply]
this commit seems to be related: https://github.com/facebook/react-native/commit/f97c6a5b498eec95e99a02c7842cb2ae160cd6cd

**Comment by myckhel** [comment]
While testing out rn v0.69.1 on android with newArchEnabled=true and hermes enabled, i got this error in metro log.
I hope i didn't miss anything.
```
 ERROR  Invariant Violation: requireNativeComponent: "RCTView" was not found in the UIManager.

This error is located at:
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:211)
    in AnimatedComponent (at createAnimatedComponent.js:264)
    in AnimatedComponentWrapper (at TouchableOpacity.js:224)
    in TouchableOpacity (at TouchableOpacity.js:280)
    in TouchableOpacity (at LearnMoreLinks.js:92)
    in RCTView (at View.js:32)
    in View (at LearnMoreLinks.js:81)
    in LinkList (at App.js:86)
    in RCTView (at View.js:32)
    in View (at App.js:69)
    in RCTView (at View.js:32)
    in View (at ScrollView.js:1682)
    in RCTScrollView (at ScrollView.js:1800)
    in ScrollView (at ScrollView.js:1826)
    in ScrollView (at App.js:65)
    in RCTView (at View.js:32)
    in View (at SafeAreaView.js:41)
    in SafeAreaView (at App.js:63)
    in App (at renderApplication.js:50)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:92)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:119)
    in AppContainer (at renderApplication.js:43)
    in AwesomeProject(RootComponent) (at renderApplication.js:60), js engine: hermes
```

**Reply by mikehardy** [reply]
If no one else has replied yet, we probably need the package.json and an App.js for reproduction

**Reply by myckhel** [reply]
Well i didnt add any package or change the content of App.js after initing the RN template.
Its same content as the one in the rn v0.69.1 template.

**Reply by mikehardy** [reply]
@myckhel interesting! That's important to know. So basically a reproduction would be:

- `npx react-native init MetroNotWorking`
-  edit android to enable new architecture, enable hermes
- `npx react-native start`
- `npx react-native run-android`

Nothing else is needed?

(https://stackoverflow.com/help/how-to-ask / https://stackoverflow.com/help/mcve)

**Reply by myckhel** [reply]
Yes correct 👍🏽

**Comment by joe-sam** [comment]
This hack fixed my new architecture build issues (using the 2022 VisualStudio developer terminal) 
From [Windows OS limit path issue](https://github.com/facebook/react-native/issues/34084)
Adding the following fixed it for me:
In the project level build.gradle
Line 1

`+ import org.apache.tools.ant.taskdefs.condition.Os`

Line 19 or after the first System.properties check (depending on your config)

```
if (System.properties['os.arch'] == "aarch64") {
   // For M1 Users we need to use the NDK 24 which added support for arch64
   ndkVersion = "24.0.8215888"
+} else if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+   // For Windows Users, we need to use NDK 23, otherwise the build will
+   // fail due to paths longer than the OS limit
+   ndkVersion = "23.1.7779620"
} else {
   // Otherwise we default to the side-by-side NDK version from AGP.
   ndkVersion = "21.4.7075529"
}

```

**Reply by mikehardy** [reply]
I wonder if, to limit the number of versions in play, window would work with the same version as aarch64 or if it must strictly be the version you specifically set here for your windows conditional branch. Have you tried it?

**Reply by joe-sam** [reply]
Yes I tried it after you suggested it and it works.
I have folded the lines mentioned above in my previous comment into this in the project level gradle

    if( (System.properties['os.arch'] == "aarch64")  || (Os.isFamily(Os.FAMILY_WINDOWS)) ){
          // For M1 Users we need to use the NDK 24 which added support for aarch64
// For Windows Users, we need to use NDK 23 +, otherwise the build will
// fail due to paths longer than the OS limit
          ndkVersion = "24.0.8215888"
      }

I also might mention that I have (at some point in the recent past) while attempting to get the new architecture build systems to work with the long paths have tweaked the LongPathsEnabled registry key in windows which circumvents the long paths issue. In fact up to now this has not worked probably due to some internal checks in the build system. Still it is probably quite important to mention.

How to change the default 256 character path length limitation (MAX_PATH) in Windows 10.

    Go to Windows Start and type REGEDIT.
    Choose the Registry Editor.
    In the Registry Editor, navigate to the following location: at HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem.
    Select the entry named: LongPathsEnabled.
    In the Value data field, enter a value of 1. This will enable the long paths option.

**Reply by mikehardy** [reply]
@cortinico something really strange happened here on this file!

- the M1 fix to use NDK24 is in (on 069-release and main)

https://github.com/facebook/react-native/commit/4befd2a29cb94b026d9c048a041aa9f1817295b5

- the Windows fix to use NDK23 *was* in, but ... was clobbered somehow and is no longer in on 069-release or main! 

commit https://github.com/facebook/react-native/commit/e48a580080bdae58b375f30fbcf8a83cc1915b2f
main https://github.com/facebook/react-native/blame/main/template/android/build.gradle
https://github.com/facebook/react-native/blob/0.69-stable/template/android/build.gradle

The git blame on that one is really strange, when going back through history, I can't figure out exactly what happened really.

Looks like 24 or a regedit (i.e., CLI doctor check in future?) might work for windows

Either way windows build broken without a fixup here *I think*

**Reply by kelset** [reply]
looks like it was removed here https://github.com/facebook/react-native/commit/7b76abc0d3a0a5bec37f314c80954e412fc5f5ec I'll ask during the sync

**Reply by mikehardy** [reply]
Oh yes - there were a series of changes here probing different styles. I may have been mentally stuck on an intermediate step. I'm certain Nicola will know much better than me here

**Reply by kelset** [reply]
yeah so the update here is that basically NDK 23 and 24 are very unstable on Win, which is why in that commit the workaround to add NDK23 was removed

**Reply by cortinico** [reply]
> I'm certain Nicola will know much better than me here

Sorry I missed this ping.
While working on Windows support, we initially thought NDK 23 would help fix the long path issues. This turned out to not be the case and we reverted the if-then-else to NDK 21 as 23/24 were really unstable on Windows and leading to a lot of Bsod.
